kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -566,31 +431,26 @@ private SnapshotImpl createSnapshot(LogSegment initSegment, Engine engine) {
    */
   private Optional<LogSegment> getLogSegmentFrom(
       Engine engine, Optional<CheckpointMetaData> startingCheckpoint) {
-    return getLogSegmentAtOrBeforeVersion(
-        engine, startingCheckpoint.map(x -> x.version), Optional.empty(), Optional.empty());
+    return getLogSegmentForVersion(
+        engine, startingCheckpoint.map(x -> x.version), Optional.empty());
   }
 
   /**
-   * Get a list of files that can be used to compute a Snapshot at or before version
-   * `versionToLoad`, If `versionToLoad` is not provided, will generate the list of files that are
-   * needed to load the latest version of the Delta table. This method also performs checks to
-   * ensure that the delta files are contiguous.
+   * Get a list of files that can be used to compute a Snapshot at version `versionToLoad`, if
+   * `versionToLoad` is not provided, will generate the list of files that are needed to load the
+   * latest version of the Delta table. This method also performs checks to ensure that the delta
+   * files are contiguous.
    *
    * @param startCheckpoint A potential start version to perform the listing of the DeltaLog,
    *     typically that of a known checkpoint. If this version's not provided, we will start listing
    *     from version 0.
-   * @param versionToLoad A specific version we try to load, but we may only load a version before
-   *     this version if this version of commit is un-backfilled. Typically used with time travel
-   *     and the Delta streaming source. If not provided, we will try to load the latest version of
-   *     the table.
+   * @param versionToLoad A specific version to load. Typically used with time travel and the Delta
+   *     streaming source. If not provided, we will try to load the latest version of the table.
    * @return Some LogSegment to build a Snapshot if files do exist after the given startCheckpoint.
    *     None, if the delta log directory was missing or empty.
    */
-  protected Optional<LogSegment> getLogSegmentAtOrBeforeVersion(
-      Engine engine,
-      Optional<Long> startCheckpoint,
-      Optional<Long> versionToLoad,
-      Optional<TableCommitCoordinatorClientHandler> tableCommitHandlerOpt) {
+  public Optional<LogSegment> getLogSegmentForVersion(
```

Comment:
why the change from `AtOrBefore` to `For`? also why `protected` -> `public`?

Reply:
With CC, it needed to get the log segment _at or before_ the version provided. Without CC we get the log segment for the version provided, and throw an error if it does not exist.

I.e. if we ask for version 3, and only version 2 is available
- with CC we return the log segment up to version 2, since there may be unbackfilled commits for version 3
- without CC we throw an error since version 3 is not available

Reply:
`protected` to `public` is because it needs to be `public` for testing. In the previous PR that added this code the package in `SnapshotManagerSuite` was modified, but without changing the filepath. I reverted that change since package should match filepath.

Reply:
Makes sense, thanks!